### PR TITLE
Feature/user class

### PR DESCRIPTION
**Pull request by jfogiato**
Files Changed: recipe.js, recipe-test.js, user.js, user-test.js

Change Summary: Created the `User` class and associated tests. Added methods to the `User` class to filter by tag and by name, as well as save/remove recipes from `user.savedRecipes`. Updated the `Recipe` class to have a `recipe.saved` property and a method to toggle this property (also added tests for this method). Refactored `recipe.listDirections` to remove the numeric identifier for each step (and instead enable this within our CSS via a <ol> tag).

Other Notes: None - feeling confident about the structure of these classes and the tests. The tests could possible use some refactoring to shift individual invocation of class methods to the `beforeEach`, but unsure on that and can be saved for later.

### CODE REVIEW

File: src/classes/Recipe.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ class Recipe {
     this.instructions = recipeData.instructions;
     this.name = recipeData.name;
     this.tags = recipeData.tags;
+    this.saved = false;
```
Comment (qrispi) [inline src/classes/Recipe.js:9]: Love this piece! Will make a lot of DOM manipulation much easier.

File: src/classes/Recipe.js [hunk 2]
```diff
@@ -46,9 +47,13 @@ class Recipe {
   }
 
   getInstructions() {
-    const instructions = this.instructions.map(step => `${step.number}. ${step.instruction}`)
+    const instructions = this.instructions.map(step => `${step.instruction}`)
```
Comment (qrispi) [inline src/classes/Recipe.js:50]: We will need to remember to add back in the list style for this piece in CSS.

File: test/Recipe-test.js [hunk 3]
```diff
@@ -282,12 +282,19 @@ describe('Recipe', () => {
...
-      "5. Bake for 9 to 10 minutes, or until you see the edges start to brown.",
-      "6. Remove the pan from the oven and let sit for 10 minutes before removing onto a cooling rack.Top with ice cream and a drizzle of chocolate sauce."
+      "In a large mixing bowl, whisk together the dry ingredients (flour, pudding mix, soda and salt). Set aside.In a large mixing bowl of a stand mixer, cream butter for 30 seconds. Gradually add granulated sugar and brown sugar and cream until light and fluffy.",
+      "Add egg and vanilla and mix until combined.",
+      "Add dry ingredients and mix on low just until incorporated. Stir in chocolate chips.Scoop the dough into 1,5 tablespoon size balls and place on a plate or sheet. Cover with saran wrap and chill at least 2 hours or overnight.When ready to bake, preheat oven to 350 degrees.",
+      "Place the cookie dough balls into ungreased muffin pan. Sprinkle with sea salt.",
+      "Bake for 9 to 10 minutes, or until you see the edges start to brown.",
+      "Remove the pan from the oven and let sit for 10 minutes before removing onto a cooling rack.Top with ice cream and a drizzle of chocolate sauce."
```
Comment (qrispi) [inline test/Recipe-test.js:290]: Whoops meant this piece in my previous comment!

**Comment by jfogiato** [comment]
Apologies for missing this piece. Closes the following issues: 

Closes #35 
Closes #36 
Closes #37 
Closes #38 
Closes #41
Closes #42
Closes #45

File: src/classes/Recipe.js (hunk 2 above)
Comment (qrispi) [reply src/classes/Recipe.js:50]: This piece is connected to the HTML we actually need to do this for.

**Review by qrispi — APPROVED** [review]
Loving the progress! We are almost there!

File: src/classes/Recipe.js (hunk 2 above)
Comment (jfogiato) [reply src/classes/Recipe.js:50]: Roger that. Issue #61 created for this.

Comment (LordSchwifty) [reply src/classes/Recipe.js:50]: Looks good!

**Comment by LordSchwifty** [comment]
Yea everything looks in order and ready to merge!

File: test/Recipe-test.js (hunk 3 above)
Comment (jsahim) [reply test/Recipe-test.js:290]: Good catch on this. Looks good to me!

**Review by jsahim — APPROVED** [review]

**Pull request merged**